### PR TITLE
Add new handler for obtaining metrics snapshot in "packets" format

### PR DESCRIPTION
**Pull request by gjoranv**
FYI: @freva, @geirst

### CODE REVIEW

File: container-core/src/main/java/com/yahoo/container/jdisc/state/MetricsPacketsHandler.java [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+        applicationName = config.application();
+    }
+
+
+    @Override
+    public ContentChannel handleRequest(Request request, ResponseHandler handler) {
+        new ResponseDispatch() {
+            @Override
+            protected Response newResponse() {
+                Response response = new Response(Response.Status.OK);
```
Comment (bjorncs) [inline container-core/src/main/java/com/yahoo/container/jdisc/state/MetricsPacketsHandler.java]: Return `405 Method Not Allowed` response for every method except GET.

File: container-core/src/main/java/com/yahoo/container/jdisc/state/MetricsPacketsHandler.java [hunk 2]
```diff
@@ -0,0 +1,188 @@
+// Copyright 2018 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+package com.yahoo.container.jdisc.state;
+
+import com.google.inject.Inject;
+import com.yahoo.collections.Tuple2;
+import com.yahoo.component.provider.ComponentRegistry;
+import com.yahoo.container.jdisc.state.StateHandler.JSONObjectWithLegibleException;
```
Comment (bjorncs) [inline container-core/src/main/java/com/yahoo/container/jdisc/state/MetricsPacketsHandler.java]: Consider moving `JSONObjectWithLegibleException` to separate file

File: container-core/src/main/java/com/yahoo/container/jdisc/state/MetricsPacketsHandler.java [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+import static com.yahoo.container.jdisc.state.StateHandler.getSnapshotPreprocessor;
```
Comment (bjorncs) [inline container-core/src/main/java/com/yahoo/container/jdisc/state/MetricsPacketsHandler.java]: Consider moving `getSnapshotPreprocessor ` to separate class

File: container-core/src/main/java/com/yahoo/container/jdisc/state/MetricsPacketsHandler.java [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+                return Collections.singleton(ByteBuffer.wrap(buildMetricOutput()));
+            }
+        }.dispatch(handler);
+
+        return null;
+    }
+
+    private byte[] buildMetricOutput() {
+        try {
+            String output = getStatusPacket() + getAllMetricsPackets();
```
Comment (bjorncs) [inline container-core/src/main/java/com/yahoo/container/jdisc/state/MetricsPacketsHandler.java]: Having multiple top-level json objects concatenated will require more complex parse login on the client side. If possible, consider wrapping all top-level objects into a single json object.

**Review by bjorncs — COMMENTED** [review]
See comments

File: container-core/src/main/java/com/yahoo/container/jdisc/state/MetricsPacketsHandler.java (hunk 4 above)
Comment (gjoranv) [reply container-core/src/main/java/com/yahoo/container/jdisc/state/MetricsPacketsHandler.java]: Sorry, we can't due to internal requirements.

File: container-core/src/main/java/com/yahoo/container/jdisc/state/MetricsPacketsHandler.java (hunk 1 above)
Comment (gjoranv) [reply container-core/src/main/java/com/yahoo/container/jdisc/state/MetricsPacketsHandler.java]: This handler is protocol agnostic (comment added).

**Review by bjorncs — APPROVED** [review]

File: container-core/src/main/java/com/yahoo/container/jdisc/state/MetricsPacketsHandler.java (hunk 2 above)
Comment (gjoranv) [reply container-core/src/main/java/com/yahoo/container/jdisc/state/MetricsPacketsHandler.java]: done.

File: container-core/src/main/java/com/yahoo/container/jdisc/state/MetricsPacketsHandler.java (hunk 3 above)
Comment (gjoranv) [reply container-core/src/main/java/com/yahoo/container/jdisc/state/MetricsPacketsHandler.java]: If there were more than one method, I would have created a common utility class or something. ATM, it looks like overkill.

**Pull request merged**